doc/migration/v5.md

```diff
@@ -0,0 +1,184 @@
+generated results of the `build_runner`.
+
+> Notice: When developing in Intellij/WebStorm it's currently also recommended to mark the directory as excluded as it 
+  will constantly change and cause Intellij to reanalyze the code frequently, which results in a loss of performance.
+
+Drop transformer related dependencies.
+
+- dart_to_js_script_rewriter
+- browser
+
+Remove transformer configuration from `pubspec.yaml`. 
+
+**Old code:**
+
+```yaml
+dependencies:
+  # ...
+
+dev_dependencies:
+  # ...
+
+transformers:
+- angular:
+    platform_directives:
+    - 'package:angular2/common.dart#COMMON_DIRECTIVES'
+    platform_pipes:
+    - 'package:angular2/common.dart#COMMON_PIPES'
+    entry_points: web/main.dart
+- test/pub_serve:
+    $include:
+      - test/**.dart
+- dart_to_js_script_rewriter
+```
+
+**New code:**
+
+```yaml
+dependencies:
+  # ...
+
+dev_dependencies:
+  # ...
+```
+
+Move your dart2js configuration out of your `pubspec.yaml` into a `build.release.yaml` config. 
+*This step is optional and the configuration may contain different configurations.*
+
+**Old code (`pubspec.yaml`):**
+
+```yaml
+transformers:
+- $dart2js:
+    checked: false
+    minify: true
+    commandLineOptions: [--show-package-warnings]
+```
+
+**New code(`build.release.yaml`):**
+
+```yaml
+targets:
+  $default:
+    builders:
+      build_web_compilers|entrypoint:
+        options:
+          compiler: dart2js
+          dart2js_args:
+          - --minify
+          - --show-package-warnings
+```
+
+### Adjust analysis_options.yaml
+
+There are several cases where `.template.dart` files are included (ex. `main.dart` and async routes). Therefore adding
+the rule `uri_has_not_been_generated` to the `analysis_options.yaml` will prevent the analyzer from reporting 
+unnecessary issues.
+
+```yaml
+linter:
+  rules:
+    - uri_has_not_been_generated
+    # ...
+``` 
+
+### Setting up the root injector and runApp
+
+angular 5.0 introduced `GenerateInjector`, which generates an injector at compile-time and as such enables better 
+tree-shaking. This requires to import the generated `.template.dart` file within the injector setup file 
+(ex. `main.dart`). 
+
+**Old code (`main.dart`)**
+
+```dart
+import 'package:my_app/app_component.dart';
+
+void main() {
+  bootstrap(AppComponent, [
+    provide(MyToken, useValue: 'test'),
+    HeroService,
+  ]);
+}
+```  
+
+**New code (`main.dart`)**
+
+```dart
+import 'package:my_app/app_component.template.dart' as ng;
+import 'main.template.dart' as self;
+
+@GenerateInjector(const [
+  const ValueProvider.forToken(MyToken, 'test'),
+  const ClassProvider(HeroService),
+])
+final InjectorFactory injectorFactory = self.injectorFactory$Injector;
+
+void main() {
+  runApp(ng.AppComponentNgFactory, createInjector: injectorFactory);
+}
+```
+
+For further information about best practices for the dependency injector see 
+[effective DI](https://github.com/dart-lang/angular/blob/master/doc/effective/di.md#injectors). 
```

Comment:
Note that if using `runApp`, then you can also add this to your `build.yaml` and it will help ddc load times for things like angular_components a lot:

```yaml
global_options:
  angular|angular:
    options:
      no-emit-component-factories: True
      no-emit-injectable-factories: True
```

Reply:
Great! I didn't know this, will check it out ASAP.

Reply:
Amazing, saved ~50% load time! Adding it to the guide.

Reply:
Nice!